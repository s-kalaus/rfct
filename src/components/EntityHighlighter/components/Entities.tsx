import React from 'react';
import { IEntity } from '~/interfaces/entity';
import { Entity } from './Entity';

export interface IProps {
  text: string;
  entities: IEntity[];
}

export const Entities = ({ text, entities }: IProps) => (
  <>
    {entities.map((entity) => (
      <Entity text={text} entity={entity} key={JSON.stringify(entity)} />
    ))}
  </>
);
