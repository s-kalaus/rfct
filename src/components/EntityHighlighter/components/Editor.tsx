import React from 'react';
import { IEntity } from '~/interfaces/entity';
import { Entities } from './Entities';
import { IStyleSheet } from '~/interfaces/style-sheet';

export interface IProps {
  text: string;
  entities: IEntity[];
  onTextChange: (text: string) => void;
  onInputRef: (node: HTMLTextAreaElement) => void;
}

export const Editor = ({
  text,
  onTextChange,
  entities,
  onInputRef,
}: IProps) => (
  <div style={styles.wrapper}>
    <textarea
      style={styles.input}
      ref={onInputRef}
      onChange={(event) => onTextChange(event.target.value)}
      value={text}
      rows={10}
      data-testid="textarea"
    />
    <Entities entities={entities} text={text} />
  </div>
);

const styles: IStyleSheet = {
  wrapper: {
    position: 'relative',
  },
  input: {
    fontFamily:
      'source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace',
    fontSize: '14px',
    background: 'none',
    border: '1px solid',
    width: '100%',
    resize: 'none',
    padding: '5px',
    whiteSpace: 'pre-wrap',
    boxSizing: 'border-box',
    wordBreak: 'break-word',
  },
};
