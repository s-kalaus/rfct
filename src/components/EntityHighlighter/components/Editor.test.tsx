import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Editor } from './Editor';
import { IProps as IEntitiesProps } from './Entities';

jest.mock('./Entities', () => ({
  Entities: (props: IEntitiesProps) =>
    `mock-entities: ${JSON.stringify(props)}`,
}));

describe('Editor', () => {
  const text = 'Text';
  const entities = [{ selectionStart: 0, selectionEnd: 0, label: 'Label' }];
  const onTextChange = jest.fn();
  const onInputRef = jest.fn();

  test('renders component', () => {
    const { container } = render(
      <Editor
        text={text}
        entities={entities}
        onTextChange={onTextChange}
        onInputRef={onInputRef}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test('react on text change', () => {
    render(
      <Editor
        text={text}
        entities={entities}
        onTextChange={onTextChange}
        onInputRef={onInputRef}
      />
    );
    userEvent.type(screen.getByTestId('textarea'), '!');
    expect(onTextChange).toHaveBeenCalledWith('Text!');
  });
});
