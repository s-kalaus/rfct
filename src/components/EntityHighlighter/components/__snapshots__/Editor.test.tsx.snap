// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor renders component 1`] = `
<div>
  <div
    style="position: relative;"
  >
    <textarea
      data-testid="textarea"
      rows="10"
      style="font-family: source-code-pro, Menlo, Monaco, Consolas, \\"Courier New\\", monospace; font-size: 14px; background: none; border: 1px solid; width: 100%; resize: none; padding: 5px; white-space: pre-wrap; box-sizing: border-box; word-break: break-word;"
    >
      Text
    </textarea>
    mock-entities: {"entities":[{"selectionStart":0,"selectionEnd":0,"label":"Label"}],"text":"Text"}
  </div>
</div>
`;
